runtime/v1/shim/service_linux.go

```diff
@@ -59,26 +63,98 @@ func (p *linuxPlatform) CopyConsole(ctx context.Context, console console.Console
 		}()
 	}
 
-	outw, err := fifo.OpenFifo(ctx, stdout, syscall.O_WRONLY, 0)
+	uri, err := url.Parse(stdout)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to parse stdout uri")
 	}
-	outr, err := fifo.OpenFifo(ctx, stdout, syscall.O_RDONLY, 0)
-	if err != nil {
-		return nil, err
+
+	switch uri.Scheme {
+	case "binary":
+		ns, err := namespaces.NamespaceRequired(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		cmd := runtime.NewBinaryCmd(uri, id, ns)
+
+		// In case of unexpected errors during logging binary start, close open pipes
+		var filesToClose []*os.File
+
+		defer func() {
+			if err != nil {
```

Comment:
I think the name `err` is confusing and not sure it is working.

common way is to name return error `retErr` in function declaration, like:

```
func dosomething() (retErr error) {
    ...
    defer func() {
         if retErr != nil {
               ...
         }
    }()
    ...
}
```

Reply:
Good to know about the convention, I was not aware. The implementation with `err` does work, you can [try it out yourself](https://play.golang.org/p/z5vp0Z0AFYw) if you like.

I've updated the declaration to specify `retErr` and replaced all references to `err` with `retErr` for consistency.

Reply:
Sorry for late reply. The link looks good. 

The convention is that we don't need `retErr` to replace existing `err`, like:

```
func doSomething() (retErr error) {
    defer func() {
         if retErr != nil {
               docleanup()
         }
    }()
    

    if err := doOther(); err != nil {
         return err
    }
    return nil
}
```

The `return err` will make `retErr = err` so that the defer function will know there is err.
It is good and easier for code reviewer 😂

Reply:
so could you revert and don't touch the existing `err`? besides from this, basically LGTM.

Thanks

Reply:
I see. I wasn't sure if `return err` would set `retErr = err` for the defer, and then I just decided to replace it in the entire function. I've reverted and updated only the declaration and defer.

Thank you for your help with this PR, I've learned a lot.